pkg/nmstate/nodenetworkstate.go

```diff
@@ -1 +1,183 @@
 package nmstate
+
+import (
+	"context"
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/golang/glog"
+
+	nmstateV1alpha1 "github.com/nmstate/kubernetes-nmstate/api/v1alpha1"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// StateBuilder provides struct for the NodeNetworkState object containing connection to
+// the cluster and the NodeNetworkState definitions.
+type StateBuilder struct {
+	Definition *nmstateV1alpha1.NodeNetworkState
+	Object     *nmstateV1alpha1.NodeNetworkState
+	apiClient  *clients.Settings
+	errorMsg   string
+}
+
+// NewStateBuilder creates a new instance of StateBuilder.
+func NewStateBuilder(apiClient *clients.Settings, name string) *StateBuilder {
```

Comment:
Do we have an scenario when we need to create NodeNetworkState manually ?

Reply:
removed